File: src/01-easy/unary/solution.swift
let msg = readLine()!
var bin = ""
for c in msg.unicodeScalars {
    let s = String(c.value, radix: 2)
    bin += String(repeating: "0", count: 7 - s.count) + s
}
var parts: [String] = []
var i = bin.startIndex
while i < bin.endIndex {
    let cur = bin[i]
    var count = 0
    while i < bin.endIndex && bin[i] == cur { count += 1; i = bin.index(after: i) }
    parts.append("\(cur == "1" ? "0" : "00") \(String(repeating: "0", count: count))")
}
print(parts.joined(separator: " "))

File: src/01-easy/unary/solution.lua
local msg = io.read()
local bin = ""
for i = 1, #msg do
    local n = msg:byte(i)
    local b = ""
    for j = 6, 0, -1 do
        b = b .. (math.floor(n / 2^j) % 2)
    end
    bin = bin .. b
end
local parts = {}
local i = 1
while i <= #bin do
    local cur = bin:sub(i, i)
    local count = 0
    while i <= #bin and bin:sub(i, i) == cur do count = count + 1; i = i + 1 end
    parts[#parts + 1] = (cur == "1" and "0" or "00") .. " " .. string.rep("0", count)
end
print(table.concat(parts, " "))
